### ISSUE TITLE
Implement Image Classification with Linear Classifiers

### ISSUE THREAD

**Issue by 5usu** [SRA-VJTI/computer-vision-study-group#4]
##  Summary
Implement **linear classifiers (SVM and Softmax)** from scratch using NumPy for image classification on CIFAR-10.  

---

##  Objective
Build intuition around:
- Linear decision boundaries
- Loss functions (hinge & softmax)
- Gradient-based optimization (SGD)
- Feature normalization and visualization

---

##  Topics Covered
- Multiclass SVM Loss & Gradient
- Softmax + Cross-Entropy Loss
- Gradient Descent / SGD
- Regularization (L2)
- Numerical Gradient Checking
- Visualizing Learned Weights

---

## Tasks
- [ ] Create a new folder: `linear_classifiers/`
- [ ] Implement:
  - `data_utils.py` — load and preprocess CIFAR-10  
  - `svm.py` — SVM loss + gradient  
  - `softmax.py` — Softmax loss + gradient  
  - `optim.py` — SGD optimizer  
  - `train.py` — training loop + accuracy calculation  
  - `grad_check.py` — numeric gradient validation  
- [ ] Train on CIFAR-10 (use 10% subset for dev)
- [ ] Visualize class weights as images (show learned templates)
- [ ] Add a README explaining key observations

---

## Directory Structure
cv_learn/
linear_classifiers/
data_utils.py
svm.py
softmax.py
optim.py
train.py
grad_check.py
README.md

---

## Tools / Dependencies
- Python 3.10+
- NumPy
- Matplotlib
- tqdm (for progress)
- CIFAR-10 dataset

---

## Example CLI
```bash
python linear_classifiers/train.py --model svm --lr 1e-3 --reg 1e-4 --epochs 20
python linear_classifiers/train.py --model softmax --lr 1e-3 --reg 5e-4 --epochs 30

**Comment by vedantmalkar** [comment]
Hello, I’d like to work on this issue could you assign it to me?

**Comment by abhixb** [comment]
hello , i would like to work on this could you please assign me
